static/app/stores/Commiters/CommittersContext.tsx

```diff
@@ -0,0 +1,274 @@
+        projectSlug: action.payload.projectSlug,
+        eventId: action.payload.eventId,
+      });
+      return {
+        ...state,
+        [key]: {
+          committers: [],
+          committersLoading: true,
+          committersError: undefined,
+        },
+      };
+    }
+    case 'set error': {
+      const key = makeCommitterStoreKey({
+        organizationSlug: action.payload.organizationSlug,
+        projectSlug: action.payload.projectSlug,
+        eventId: action.payload.eventId,
+      });
+
+      return {
+        ...state,
+        [key]: {
+          committers: [],
+          committersLoading: false,
+          committersError: true,
+        },
+      };
+    }
+    case 'reset': {
+      return {};
+    }
+    default: {
+      exhaustive(action);
+      return state;
+    }
+  }
+}
+
+export const CommittersContext = createContext<
+  [CommittersState, React.Dispatch<CommittersAction>] | null
+>(null);
+
+interface CommittersContextProviderProps {
+  children: React.ReactNode;
+  initialState?: CommittersState;
+}
+
+export function CommittersProvider(props: CommittersContextProviderProps) {
+  const contextValue = useReducer(committersReducer, props.initialState ?? {});
+
+  return (
+    <CommittersContext.Provider value={contextValue}>
+      {props.children}
+    </CommittersContext.Provider>
+  );
+}
+
+export function useCommitters(): [CommittersState, React.Dispatch<CommittersAction>] {
+  const context = useContext(CommittersContext);
+
+  if (!context) {
+    throw new Error('useCommitters called outside of CommittersContext.Provider');
+  }
+
+  return context;
+}
+
+interface RequiredWithCommittersProps {
+  event: Event;
+  organization: Organization;
+  project: Project | AvatarProject;
+  group?: Group;
+}
+
+export interface WithCommittersProps {
+  committers: Committer[] | undefined;
+}
+
+export function withCommitters<P extends WithCommittersProps>(
+  Component: React.ComponentType<P>
+) {
+  const wrappedComponent: React.ComponentType<
+    Omit<P, keyof WithCommittersProps> & RequiredWithCommittersProps
+  > = (props): React.ReactElement => {
+    const api = useApi();
+    const [state, dispatch] = useCommitters();
+
+    const key = useMemo(() => {
+      return makeCommitterStoreKey({
+        organizationSlug: props.organization.slug,
+        projectSlug: props.project.slug,
+        eventId: props.event.id,
+      });
+    }, [props.organization.slug, props.project.slug, props.event.id]);
+
+    useEffect(() => {
+      if (!props.group?.firstRelease) {
+        return undefined;
+      }
+
+      if (state[key]?.committers || state[key]?.committersLoading) {
+        return undefined;
+      }
+
+      dispatch({
+        type: 'start loading',
+        payload: {
+          eventId: props.event.id,
+          organizationSlug: props.organization.slug,
+          projectSlug: props.project.slug,
+        },
+      });
+
+      let unmounted = false;
+
+      fetchCommitters(api, {
```

Comment:
@k-fish yes you are correct. I think this is what the code was already doing before if I'm not mistaken. In practice, I think this is unlikely to be a problem (altho that's a poor excuse).

In the example you described, I think it'd depend on which props were to change. I would expect that in our runtime code, props like organization, project and event are all stable object references so that what you described isn't an issue and we dont reload data when we dont expect the response to change (the useeffect deps).